defaultclient.go

```diff
@@ -133,6 +136,15 @@ func (client *DefaultClient) ValidatePermission(claims *JWTClaims, requiredPermi
 		return true, nil
 	}
 
+	// if the request is coming from a client (i.e. no claims.sub), we want to check using the client's permissions and
+	// organizationID
+	if claims.Subject == "" {
+		client.applyClientPermissionResourceValues(claims.Permissions, claims.OrganizationID)
+		if client.permissionAllowed(claims.Permissions, requiredPermission) {
+			return true, nil
+		}
+	}
```

Comment:
Not sure if this is the right way -- in AGS, `{namespace}` is replaced immediately with the `claims.namespace` when the token is granted. For example:

```json
{
  "clientId": "a42723e41d2d4dda90974c019461919b",
  "namespace": "foundations",
  // ...
  "clientPermissions": [
    // ...
    { "resource": "ADMIN:NAMESPACE:{namespace}:HEHE", "action": 2 }
  ]
}
```

Request:

```
curl --location 'https://internal.gamingservices.accelbyte.io/iam/v3/oauth/token' \
--header 'Content-Type: application/x-www-form-urlencoded' \
--header 'Authorization: ••••••' \
--data-urlencode 'grant_type=client_credentials'

{
  "access_token": "<redacted>",
  // ...
  "namespace": "foundations",
  "permissions": [
      // ...
      {
          "resource": "ADMIN:NAMESPACE:foundations:HEHE",
          "action": 2
      }
  ]
}
```

So probably IC service is where we want to change it (sorry, it was my mistake I didn't check the AGS behavior first). What do you think @jordan-ab?

Reply:
I think you're right - after looking through the IAM codebase it looks like it fills the placeholder with the appropriate namespace at the time of generating the token. I'll do the same with IC service :+1: